IndicoV2.Abstractions/Storage/IStorageClient.cs

```diff
@@ -13,7 +13,7 @@ public interface IStorageClient
         Task<Stream> GetAsync(Uri uri, CancellationToken cancellationToken);
 
         [Obsolete("On this level it's preferable to use streams instead of paths")]
-        Task<IEnumerable<IFileMetadata>> UploadAsync(IEnumerable<string> filePaths, CancellationToken cancellationToken);
+        Task<IEnumerable<IFileMetadata>> UploadAsync(IEnumerable<string> filePaths, CancellationToken cancellationToken, int batchSize);
 
         Task<(string Name, string Meta)[]> UploadAsync(IEnumerable<(string Path, Stream Content)> files,
```

Comment:
This needs a batch -- since they're sending files and streams of content.

Reply:
I'm a little unsure how we would batch here: https://github.com/IndicoDataSolutions/indico-client-csharp/pull/142/files#diff-9ff35bbc87d0ce2f5d124cc4fa4c0f4ffb1b91db080e44edd79992d28c42e17aR87

Looks like it's just iterating and writing to a stream per file. Can you write multiple files to a stream in a batch? Would that save time/affect performance?

Reply:
You could intervene at a level higher than that (like how you've got CallBatched for the filepaths version). What the function you're looking at is doing is taking *all* of the streams, writing them to a multipart form thing, and then uploading them together (so if you send 100 streams that would dump all of them into one request together as if it were 100 files)

So possibly here would be a good spot to do something similar in terms of batching the calls by divvying up the `IEnumerable` into chunks and uploading each chunk and then merging the results. 
https://github.com/IndicoDataSolutions/indico-client-csharp/blob/3319e3f8d40fba1721ea0081ec8914de1d11a3d2/IndicoV2/Storage/StorageClient.cs#L69

Reply:
ah okay thanks a lot that makes sense now